stories/Button/Guideline.mdx

```diff
@@ -0,0 +1,169 @@
+import {
+  Meta,
+  Story,
+  Canvas,
+  Primary,
+  Controls,
+  Unstyled,
+  Source,
+} from '@storybook/blocks';
+import { Button } from '../../src/lib/components/buttonv2/Buttonv2.component';
+
+import * as ButtonStories from './button.stories';
+
+<Meta of={ButtonStories} />
+
+# Button
+
+Buttons are used to trigger an action or event when clicked.
+
+## Size & style
+
+### Style
+
+- Height: 2rem
+- Border-radius: 3px
+- Minimum width: 5rem
+
+### Padding
+
+Button’s label measurement:
+
+- Line height: 1.25rem
+- Vertical padding: 0.5rem
+- Horizontal padding of 1rem
+- Space between icon & label: 0.5rem
+
+### Button spacing
+
+- Horizontal spacing: 1rem
+- Vertical spacing: 0.75rem
+- Space between 2 buttons: 1.5rem - when there is no space constraint
+
```

Comment:
@Cuervino shall we add a styling rule when we have icons in the button ?
Maybe also a behavior rule for the "loading" buttons where we replace the icon by some spinners

Reply:
@JBWatenbergScality 
1. what do you mean by styling rules?
2. Agree for the loading rules, though I don't know them.

Reply:
For styling: I mean spacing between text and icon, baseline alignements, impact on accessibility. Can we have 2 icons on a single button (at left and right of the text, probably this shall be forbidden ), where do we prefer to put the icon (left or right of the text?, for the right case it seems to be only valid for the 'continue' button with the right arrow) 

For behavior:
- loading behavior
- disabled behavior
- if there is an icon next to the text is the icon read outloud by screen readers in addition to the text ?

Reply:
My proposal

Guidelines that could be added:

---

### Icon-Text Spacing
- **Spacing**: 0.5rem between the text and the icon.

### Icon Usage
- **Placement**: Icons are to be positioned only on the left side of the text.
- **Alignment**: Icons should be vertically center-aligned with the text.

### Accessibility
- **Icons in Button Design**: 
  - Easily recognizable icons help with the quick recognition of a button’s function. If the icon is not clear enough, it loses its purpose, so in such cases, avoid using icons.
  
- **Aria Labels**: 
  - Providing descriptive labels for screen readers is really helpful, especially for buttons that use icons only.

### Button States

#### Loading State
- Display a spinner animation within the button.
- Disable any interactions during the loading process.

#### Disabled State
- Clicks on the button should trigger no action, and the cursor should display a "not-allowed" indicator.
- A tooltip explaining the reason for the button’s disabled state should appear on hover.


---

Note:
The loading state behaviour should be described in the story.